app/vmctl/remoteread.go

```diff
@@ -125,3 +146,59 @@ func (rrp *remoteReadProcessor) do(ctx context.Context, filter *remoteread.Filte
 		return nil
 	})
 }
+
+func remoteReadImport([]string) {
+	ctx, cancel := context.WithCancel(context.Background())
+	signalHandler(cancel)
+
+	if *remoteReadSrcAddr == "" {
+		logger.Fatalf("flag --remote-read-src-addr cannot be empty")
+	}
+	if *remoteReadStepInterval == "" {
+		logger.Fatalf("flag --remote-read-step-interval cannot be empty")
+	}
+
+	rr, err := remoteread.NewClient(remoteread.Config{
+		Addr:               *remoteReadSrcAddr,
+		Username:           *remoteReadUser,
+		Password:           *remoteReadPassword,
+		Timeout:            *remoteReadHTTPTimeout,
+		UseStream:          *remoteReadUseStream,
+		Headers:            *remoteReadHeaders,
+		LabelName:          *remoteReadFilterLabel,
+		LabelValue:         *remoteReadFilterLabelValue,
+		InsecureSkipVerify: *remoteReadInsecureSkipVerify,
+	})
+	if err != nil {
+		logger.Fatalf("error create remote read client: %s", err)
+	}
+
+	vmCfg := initConfigVM()
+
+	importer, err := vm.NewImporter(vmCfg)
+	if err != nil {
+		logger.Fatalf("failed to create VM importer: %s", err)
+	}
+
+	go func() {
```

Comment:
why do not use `defer importer.Close()`? 

 ```
if err := ctx.Err(); err != nil {
			logger.Errorf("context cancel err: %s\n", err)
		}
```

 Hm, I think it always return an error, if user interrupts vmctl with `ctrl^C` command, but it's not really an error. Maybe we should skip this check?


 Also original code waits for sigint and cancels context:
```go
signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
	go func() {
		<-c
		fmt.Println("\r- Execution cancelled")
		if importer != nil {
			importer.Close()
		}
		cancelCtx()
	}()
```

 It could be rewritten for

```go
        # rmp.run() 
        # must be unblock-able and start background process
        if err := rmp.run(ctx, *globalSilent, *globalVerbose); err != nil {
		logger.Fatalf("error start run import via remote read protocol: %s", err)
	}
        # wait for terminate command at main thread
        sig := procutil.WaitForSigterm()
	logger.Infof("service received signal %s", sig)
        # cancel run context
	cancel()
        # wait at main thread for proper program stop
        importer.Close()
```

Reply:
discussed, fixed context usage